libhoth_spi.c

```diff
@@ -195,10 +202,15 @@ int libhoth_spi_open(const struct libhoth_spi_device_init_options* options,
 
   spi_dev->fd = fd;
   spi_dev->mailbox_address = options->mailbox;
-  spi_dev->address_mode_4b = 1;
+  spi_dev->address_mode_4b = true;
```

Comment:
nit: if we're always assuming 4B, it might be worthwhile to remove and simplify the function signatures.

Not required for this PR since it's cleanup I can pick up.

Reply:
Actually, the opposite: the user ought to be able to enable this partially-implemented feature; it's merely coincidental that all our current uses of the spidev transport use 4-byte address mode exclusively.